api/src/controllers/upgrade.js

```diff
@@ -1,40 +1,62 @@
 const auth = require('../auth');
 const serverUtils = require('../server-utils');
 
-const service = require('../services/upgrade');
+const service = require('../services/setup/upgrade');
 const configWatcher = require('../services/config-watcher');
 
+const REQUIRED_PERMISSIONS = ['can_configure'];
```

Comment:
I think there already is an issue scheduled for `3.15`: https://github.com/medic/cht-core/issues/7139
Do you think we should move that to 4.0?

Reply:
Ahh that makes sense. Let's leave it in 3.15 for now. It could be considered a breaking change because users will lose access to that page, but I suspect everyone upgrading is a full admin anyway...